flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/typeutils/BinaryGenericSerializer.java

```diff
@@ -127,63 +133,41 @@ public int hashCode() {
 
 	@Override
 	public TypeSerializerSnapshot<BinaryGeneric<T>> snapshotConfiguration() {
-		return new BinaryGenericSerializerSnapshot<>(serializer);
+		return new BinaryGenericSerializerSnapshot<>(this);
+	}
+
+	public TypeSerializer<T> getInnerSerializer() {
+		return serializer;
 	}
 
 	/**
 	 * {@link TypeSerializerSnapshot} for {@link BinaryGenericSerializer}.
 	 */
-	public static final class BinaryGenericSerializerSnapshot<T> implements TypeSerializerSnapshot<BinaryGeneric<T>> {
-		private static final int CURRENT_VERSION = 3;
-
-		private TypeSerializer<T> previousSerializer;
+	public static final class BinaryGenericSerializerSnapshot<T> extends CompositeTypeSerializerSnapshot<BinaryGeneric<T>, BinaryGenericSerializer<T>> {
```

Comment:
Since the type of snapshot is changed, is this a break change to existing checkpoint?

Reply:
Yes...